### ISSUE TITLE
hphosts now uses https

### ISSUE THREAD

**Issue by TheRyuu** [gorhill/uBlock#1749]
The site now uses https and redirects http to https[1].  This apparently had the side effect of breaking its usage in uBlock as it now shows as 0 out of 0 filters even when forcing an update.

The filter lists should be updated to use https in both uBlock and uMatrix.

Perhaps this also does raise another point, if the site redirects http to https, should uBlock be resilient and allow this?  It seems uMatrix continued to work correctly.

[1] https://hphosts.blogspot.com/2016/06/info-hphosts-forums-and-website.html

**Comment by lewisje** [comment]
I am now curious about the best practices for supporting a 301; FTR, the original URL is http://hosts-file.net/.%5Cad_servers.txt and these are my headers...

`Connect to 107.22.171.143 on port 80 ... ok`

```
GET /.%5Cad_servers.txt HTTP/1.1
Host: hosts-file.net
Connection: close
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36
Accept-Encoding: gzip
Accept-Charset: ISO-8859-1,UTF-8;q=0.7,*;q=0.7
Cache-Control: no-cache
Accept-Language: de,en;q=0.7,en-us;q=0.3
Referer: http://web-sniffer.net/
```

and below is the response:

```
Status: HTTP/1.1 301 Moved Permanently
Content-Length: 160
Content-Type: text/html; charset=UTF-8
Location: https://hosts-file.net/ad_servers.txt
Server: Microsoft-IIS/7.5
X-Powered-By: ASP.NET
Date: Tue, 28 Jun 2016 17:17:08 GMT
Connection: close
```

```html
<head><title>Document Moved</title></head>
<body><h1>Object Moved</h1>This document may be found <a HREF="https://hosts-file.net/ad_servers.txt">here</a></body>
```

I'm surprised the 301 page didn't include a META refresh and a JS redirection for good measure.

**Comment by migueldemoura** [comment]
I'm not getting the 301, they seem to have removed it - access to that file can be done with http or https.
List should get update to use https, though.

**Comment by RacerDuke** [comment]
Doh! I accidentally posted a duplicate issue under uAssets. But yes, I also support updating the entry to use https.

**Comment by gorhill** [comment]
Fixed with 3b9fd49c508812dfe2545836c63bd5d8cfd2ead4.
